### PR TITLE
Code snippet variables

### PR DESCRIPTION
**Pull request by mheap**
## Description

We don't have much consistency on variable name usage and descriptions. This PR converts the free text table to a list of names that pull the descriptions from a config file.

If we agree that this is a good idea, I'm happy to update all of the existing snippets

## Deploy Notes

N/A

### CODE REVIEW

**Comment by benhayehudi** [comment]
Would it be possible to make these changes in the new markdown renderer gem as that will soon become the source of the Markdown Pipeline?

**Comment by mheap** [comment]
@benhayehudi I saw that the PR to merge that was closed. I'm happy to move the filter across later when we're ready to revisit that pipeline

**Comment by benhayehudi** [comment]
The PR to merge it into NDP was closed, but the work on the renderer is almost done in its repo.

File: spec/filters/snippet_variables_filter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe SnippetVariablesFilter do
+  it 'returns whitespace with whitespace input' do
```
Comment (fabianrbz) [inline spec/filters/snippet_variables_filter_spec.rb]: you can place the strings in the method call and expectation and gert rid of the variables

**Comment by lornajane** [comment]
Can we update the instructions as well please since lots of "outsiders" contribute to snippets as well? Also is it possible to override this if something different is needed in one particular code snippet context, or should I use a different variable name in that situation so that I can control the copy?

**Comment by mheap** [comment]
@lornajane What kind of situation are you seeing where we'd need the same variable name but a different description?

**Comment by lornajane** [comment]
We mostly workaround this by using a selection of `NUMBER`, `TO_NUMBER`, `RECIPIENT_NUMBER` and that sort of thing but for example in SMS API we use

```
TO_NUMBER | The phone number you are sending the message to.
```

But in Voice API that is instead

```
TO_NUMBER | The number you would like to call to in E.164 format. For example 447700900001.
```

I quite like that we describe what the number is used for _in this code snippet_ and while I am totally +1 for things like API keys/secrets, for the variables I might like to be descriptive within the specific context where I'm using the field - if that makes any sense!

**Comment by mheap** [comment]
It makes total sense. How about being explicit in what's required, but allowing us to render in a certain way? e.g.

```
SMS_TO_NUMBER:
  description: The number you are sending the SMS to in E.164 format. For example `447700900000`.
  render_as: TO_NUMBER
VOICE_TO_NUMBER:
  description: The number you would like to call to in E.164 format. For example 447700900001.
  render_as: TO_NUMBER
```

This would allow us to have a central source of variable names to improve consistency whilst still allowing for additional information based on the context

**Comment by lornajane** [comment]
That would work - I think it's either some sort of renaming dodge like this, or per-product configuration which would encourage us to stay consistent within each section and not lose our way with what could become quite a long list eventually. I'm for per-product but it's not me that is going to build it so absolutely your call.

**Comment by mheap** [comment]
I could go either way on it.

@tbedford As the main consumer of this, would you prefer one big list with `render_as` or multiple files, and specifying the one to use in the snippet e.g.


````
```snippet_variables
type: voice
variables:
  - TO_NUMBER
  - FROM_NUMBER
````

**Comment by dragonmantank** [comment]
I would say, as someone who has to write the code snippets (and would like to automate running them...) having per-product names would be nice, but I totally understand if that's not maintainable.

**Comment by tbedford** [comment]
@mheap I would prefer one file with everything in it, so the `render_as` option.

**Comment by mheap** [comment]
@dragonmantank Would you be able to run your tooling with a single file and `render_as`?

**Comment by dragonmantank** [comment]
@mheap I'd rather not have to pull down the code standards repo just to run tests, just have a clear naming convention for this (though, honestly, the current runner doesn't care since you can embed environment variables into the test, it's more of an issue if you want to have local values for testing through something like a `.env` file).

**Comment by mheap** [comment]
@dragonmantank If it's contained in a single file we can copy between NDP and the snippet tester. Not an ideal solution, but the rate of change for the variable change is intended to be small so it should work for now.

I'll work on updating this PR to use the new Markdown pipeline and variable names file

**Pull request closed without merging**